src/services/pathCompletions.ts

```diff
@@ -312,9 +312,11 @@ namespace ts.Completions.PathCompletions {
 
     function getCompletionEntriesFromTypings(host: LanguageServiceHost, options: CompilerOptions, scriptPath: string, span: TextSpan, result: CompletionEntry[] = []): CompletionEntry[] {
         // Check for typings specified in compiler options
+        const seen = createMap<true>();
         if (options.types) {
-            for (const moduleName of options.types) {
-                result.push(createCompletionEntryForModule(moduleName, ScriptElementKind.externalModuleName, span));
+            for (const typesName of options.types) {
+                const moduleName = getPackageNameFromAtTypesDirectoryWithoutPrefix(typesName);

```

Comment:
Nit: don't really need the local

Reply:
True, but I found it helpful for debugging.

Reply:
If debugging in chrome you can actually see the return value before exiting a function even if it's not named. That's in "Scope > Local > Return value" in the right-hand menu.